bin/verify-sql-formatting.py

```diff
@@ -0,0 +1,104 @@
+import argparse
+import difflib
+import os
+import re
+import shutil
+import sqlparse
+
+
+def print_without_common_lines(original, parsed):
+    d = difflib.Differ()
+    diff = d.compare(
+        original.splitlines(keepends=True), parsed.splitlines(keepends=True)
+    )
+    buffer = []
+    counter = 1
+    for line in diff:
+        if line.startswith(" "):
+            counter += 1
+        if line.startswith("-"):
+            buffer.append(f"Line {counter}:")
+            counter += 1
+        if line.startswith("-") or line.startswith("+") or line.startswith("?"):
+            buffer.append(repr(line))
+    print("\n".join(buffer))
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--path", "-p", type=dir_path, required=True, help="Parsing for sql files in this directory and its subdirectories.")
+    parser.add_argument("--backup", "-b", action="store_true", help="Backup files before processing.")
+    parser.add_argument("--restore-backup", "-r", action="store_true", help="Restore files from backup, no other actions are run")
+    parser.add_argument("--update", "-u", action="store_true", help="Overwrite sql files with new formatting")
+    parser.add_argument("--dry-run", "-d", action="store_true", help="Only show changes, but do not write to files")
+
+    return parser.parse_args()
+
+
+def dir_path(path):
+    if os.path.isdir(path):
+        return path
+    else:
+        raise argparse.ArgumentTypeError(f"readable_dir:{path} is not a valid path")
+
+
+def parse_sqlite_file(original):
+    #sqlparse cannot parse sqlite files, and might mangle comments
+    safeguard_sqlite_info = []
+    for line in original.splitlines():
+        if re.match(r'^[ ]*.|(--)', line):
+            safeguard_sqlite_info.append("/*TEMP" + line + "TEMP*/")
```

Comment:
Does this break on trailing comments? Does this mean any line starting with a space is assumed to be a comment?

Reply:
I should escape the dot.
Trailing comments stay trailing with the normal parser, but single-line comments would be added after the last SQL statement, that did not look right for me, so I changed it.